app/src/hid.c

```diff
@@ -122,13 +184,33 @@ static inline bool check_keyboard_usage(zmk_key_t usage) {
         }                                                                                          \
     }
 
+#if IS_ENABLED(CONFIG_ZMK_USB_BOOT)
+static zmk_hid_boot_report_t *get_boot_report() {
+    HID_CHECK_ROLLOVER_ERROR();
+#if CONFIG_ZMK_HID_KEYBOARD_REPORT_SIZE != HID_BOOT_KEY_LEN
+    boot_report.modifiers = keyboard_report.body.modifiers;
+    int len = CONFIG_ZMK_HID_KEYBOARD_REPORT_SIZE < HID_BOOT_KEY_LEN ? CONFIG_ZMK_HID_KEYBOARD_REPORT_SIZE : HID_BOOT_KEY_LEN;
+    memcpy(&boot_report.keys, keyboard_report.body.keys, len);
```

Comment:
If using memcpy, it would also be a good idea to `BUILD_ASSERT(sizeof(boot_report.keys[0]) == sizeof(keyboard_report.body.keys[0]), ...)` to catch if someone changes these types to be incompatible.